crates/re_renderer/src/wgpu_resources/render_pipeline_pool.rs

```diff
@@ -102,23 +101,40 @@ pub struct RenderPipelineDesc {
     pub multisample: wgpu::MultisampleState,
 }
 
+#[derive(thiserror::Error, Debug)]
+pub enum RenderPipelineError {
+    #[error("referenced pipeline layout not found")]
+    PipelineLayoutNotPresent(),
+
+    #[error("referenced vertex shader not found")]
+    VertexShaderNotFound(),
+
+    #[error("referenced fragment shader not found")]
+    FragmentShaderNotFound(),
+}
+
 impl RenderPipelineDesc {
     fn create_render_pipeline(
         &self,
         device: &wgpu::Device,
         pipeline_layouts: &GpuPipelineLayoutPool,
         shader_modules: &GpuShaderModulePool,
-    ) -> anyhow::Result<wgpu::RenderPipeline> {
-        let pipeline_layout = pipeline_layouts
-            .get_resource(self.pipeline_layout)
-            .context("referenced pipeline layout not found")?;
-
-        let vertex_shader_module = shader_modules
-            .get(self.vertex_handle)
-            .context("referenced vertex shader not found")?;
-        let fragment_shader_module = shader_modules
-            .get(self.fragment_handle)
-            .context("referenced fragment shader not found")?;
+    ) -> Result<wgpu::RenderPipeline, RenderPipelineError> {
+        let pipeline_layout =
+            if let Ok(pipeline_layout) = pipeline_layouts.get_resource(self.pipeline_layout) {
+                Some(&pipeline_layout.layout)
+            } else {
+                return Err(RenderPipelineError::PipelineLayoutNotPresent());
+            };
+
+        let Ok(vertex_shader_module) = shader_modules.get(self.vertex_handle) else {
+            re_log::error!("referenced vertex shader not found");
+            return Err(RenderPipelineError::VertexShaderNotFound());
```

Comment:
Don't both log AND return the error - that will likely get it logged twice!

See https://github.com/rerun-io/rerun/blob/main/CODE_STYLE.md#error-handling-and-logging for more